.github/workflows/preview.yml

```diff
@@ -0,0 +1,40 @@
+name: Amplify Preview
+
+on:
+  pull_request:
+    types: [labeled]
+
+jobs:
+  deploy:
+    if: contains(github.event.pull_request.labels.*.name, 'Preview Environment')
+    runs-on: ubuntu-latest
+    continue-on-error: true
+    permissions:
+      id-token: write
+      pull-requests: write
+    steps:
+      - uses: actions/checkout@v3
+      - name: configure AWS credentials
+        id: creds
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-region: us-east-1
+          role-to-assume: ${{ secrets.AWS_GITHUB_ACTIONS_ROLE }}
+          output-credentials: true
+      - name: set branch name env
+        id: setbranchname
+        run: |
+          # use GITHUB_HEAD_REF that is set to PR source branch
+          echo "branch=$GITHUB_HEAD_REF" >> "$GITHUB_OUTPUT"
```

Comment:
If you make a follow-up PR, I think this step can be removed in both workflows.

Reply:
Oh, thought I did that already, must have come back in rebase.